transport-native-epoll/src/main/java/io/netty/channel/epoll/AIOContext.java

```diff
@@ -62,34 +70,54 @@ public long getAddress() {
         assert !destroyed;
         assert dst.isDirect();
         assert dst.position() == 0;
+        assert file.epollEventLoop.aioContext == this;
+
+        int length = (dst.limit() & 511) == 0 ? dst.limit() : ((dst.limit() + 511) & ~511);
```

Comment:
nit: This line could be moved in a separate method, explaining its purpose (e.g. `getSmallestAlignedBoundary(ByteBuffer, int)`) and parametrizing the value of 511 or 512.

Reply:
There used to be a check on `file.isDirect()` that is now missing, is this intended? I also agree that the part of code that deals with alignment is probably better moved to a dedicated method to help with readability, not just the calculation, the 2 checks too.

Reply:
It does check if the buffer is direct, sure I can make the checks a sep method